src/support.cpp

```diff
@@ -1272,17 +1272,19 @@ void AreaSupport::moveUpFromModel(const SliceDataStorage& storage,
  */
 std::pair<Polygons, Polygons> AreaSupport::computeBasicAndFullOverhang(const SliceDataStorage& storage, const SliceMeshStorage& mesh, const unsigned int layer_idx)
 {
-    Polygons supportLayer_supportee = mesh.layers[layer_idx].getOutlines();
+    Polygons outlines = mesh.layers[layer_idx].getOutlines();
     constexpr bool no_support = false;
     constexpr bool no_prime_tower = false;
-    Polygons supportLayer_supporter = storage.getLayerOutlines(layer_idx - 1, no_support, no_prime_tower);
+    Polygons outlines_below = storage.getLayerOutlines(layer_idx - 1, no_support, no_prime_tower);
+    constexpr int layers_above = 3; // this is now a random value that seems to work fine. Do we need to expose this in the front end?
+    Polygons outlines_above = layer_idx + layers_above < mesh.layers.size() ? storage.getLayerOutlines(layer_idx + layers_above, no_support, no_prime_tower) : Polygons();
 
     const coord_t layer_height = mesh.settings.get<coord_t>("layer_height");
     const AngleRadians support_angle = mesh.settings.get<AngleRadians>("support_angle");
     const double tan_angle = tan(support_angle) - 0.01; // The X/Y component of the support angle. 0.01 to make 90 degrees work too.
     const coord_t max_dist_from_lower_layer = tan_angle * layer_height; // Maximum horizontal distance that can be bridged.
-    Polygons supportLayer_supported = supportLayer_supporter.offset(max_dist_from_lower_layer);
-    Polygons basic_overhang = supportLayer_supportee.difference(supportLayer_supported);
+
+    Polygons basic_overhang = outlines.difference(outlines_below.offset(max_dist_from_lower_layer)).intersection(outlines_above.offset(-max_dist_from_lower_layer * layers_above));
```

Comment:
Add comments and spread over multiple lines. Please check my comments if they make sens!
```suggestion
    Polygons basic_overhang = outlines
        // overhang areas protruding less then `max_dist_from_lower_layer` don't need support
        .difference(outlines_below.offset(max_dist_from_lower_layer))
        // limit intersection to the offsetted outlines above; this way a overhang has to exist for _at least_ `layers_above` number of layers before support is generated
        .intersection(outlines_above.offset(-max_dist_from_lower_layer * layers_above));
```